modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4TrashingAllocatorIT.java

```diff
@@ -0,0 +1,114 @@
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestHandler;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+public class Netty4TrashingAllocatorIT extends ESNetty4IntegTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return CollectionUtils.concatLists(List.of(Handler.class), super.nodePlugins());
+    }
+
+    @Override
+    protected boolean addMockHttpTransport() {
+        return false;
+    }
+
+    public void testTrashContent() throws InterruptedException {
+        try (var client = new Netty4HttpClient()) {
+            var addr = randomFrom(internalCluster().getInstance(HttpServerTransport.class).boundAddress().boundAddresses()).address();
+            var responses = client.post(addr, List.of(new Tuple<>(Handler.ROUTE, randomAlphaOfLength(between(1024, 2048)))));
+            assertEquals(HttpResponseStatus.OK, responses.stream().findFirst().get().status());
+        }
+    }
+
+    public static class Handler extends Plugin implements ActionPlugin {
+        static final String ROUTE = "/_test/trashing-alloc";
+
+        @Override
+        public Collection<RestHandler> getRestHandlers(
+            Settings settings,
+            NamedWriteableRegistry namedWriteableRegistry,
+            RestController restController,
+            ClusterSettings clusterSettings,
+            IndexScopedSettings indexScopedSettings,
+            SettingsFilter settingsFilter,
+            IndexNameExpressionResolver indexNameExpressionResolver,
+            Supplier<DiscoveryNodes> nodesInCluster,
+            Predicate<NodeFeature> clusterSupportsFeature
+        ) {
+            return List.of(new BaseRestHandler() {
+                @Override
+                public String getName() {
+                    return ROUTE;
+                }
+
+                @Override
+                public List<Route> routes() {
+                    return List.of(new Route(RestRequest.Method.POST, ROUTE));
+                }
+
+                @Override
+                protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+                    var iter = request.releasableContent().iterator();
+                    return (chan) -> {
+                        request.getHttpRequest().release();
+                        var contentTrashed = true;
+                        BytesRef br;
+                        while ((br = iter.next()) != null && contentTrashed) {
+                            for (int i = br.offset; i < br.offset + br.length; i++) {
+                                if (br.bytes[i] != 0) {
+                                    contentTrashed = false;
```

Comment:
nit: why not throw an `AssertionError` right here, rather than converting this into a 500 and then asserting that the HTTP response is a 200?

Reply:
Since it's an integ test I prefer asserting in/out from the client side. I think it's easier to read from the test method what it does, in general, may be not this particular case. No strong opinion about this particular case.